src/script/interpreter.cpp

```diff
@@ -1252,6 +1281,54 @@ PrecomputedTransactionData::PrecomputedTransactionData(const T& txTo)
 template PrecomputedTransactionData::PrecomputedTransactionData(const CTransaction& txTo);
 template PrecomputedTransactionData::PrecomputedTransactionData(const CMutableTransaction& txTo);
 
+template <class T>
+uint256 SignatureHashOutput(const CScript& scriptCode, const T& txTo, unsigned int nOut, int nHashType, const CAmount& amount, SigVersion sigversion, const PrecomputedTransactionData* cache)
+{
+    assert(nOut < txTo.vout.size());
+
+    uint256 hashPrevouts;
+    uint256 hashSequence;
+    uint256 hashOutputs;
+    const bool cacheready = cache && cache->ready;
+
+    if (!(nHashType & SIGHASH_ANYONECANPAY)) {
+        hashPrevouts = cacheready ? cache->hashPrevouts : GetPrevoutHash(txTo);
```

Comment:
This appears to differ from the qip-5 specification as an ANYONECANPAY signature should sign the first vin. Currently, the signature produced here would allow replaying an ANYONECANPAY OP_SENDER output in another transaction.

Reply:
Include the prevout and sequence hash for the first VIN in case of the flag ANYONECANPAY